geometry/meshop.hpp

```diff
@@ -429,6 +429,22 @@ void saveAsPly( const Mesh::pointer &mesh
 void saveAsPly( const Mesh &mesh
                , const boost::filesystem::path &filepath);
 
+/** Save mesh as PLY, optionally include vertex colors, normals or face colors.
+ *
+ * \param mesh mesh to save
+ * \param filepath path to output file
+ * \param vertexColors vertex colors, pass empty if no colors are to be saved
+ * \param faceColors face colors, pass empty if no colors are to be saved
+ * \param vertexNormals vertex normals, pass empty if no normals are to be saved
+ * \param faceLabels face labels, pass empty if no labels are to be saved
+ */
+void saveAsPlyWithFeatures(const boost::filesystem::path& filepath,
+                           const Mesh& mesh,
+                           const math::Points3i& vertexColors = {},
+                           const math::Points3i& faceColors = {},
+                           const math::Points3& vertexNormals = {},
+                           const std::vector<int>& faceLabels = {});
```

Comment:
`optional` is semantically better than an empty vector I would say.

Reply:
I discussed this with @pavelsevecek - he suggested that empty lists are better than optional references and I agree with his opinion.

Reply:
For context, optional references are a bit controversial, that's why `std::optional` cannot store references (yet). We can do that with `boost::optional`, but I'd rather avoid doing so if possible.
See also https://www.fluentcpp.com/2018/10/05/pros-cons-optional-references/

Reply:
What about an OOP solution; e.g. creating a wrapper class that would contain the geometry::Mesh and all its optional "features"? Would greatly simplify the function signature for saving/loading mesh with features. @zap150

Reply:
Closing as the function was moved to an internal module.